docs/DeveloperGuide.md

```diff
@@ -4,26 +4,231 @@
+
+2. **Find by ISBN**
+    - Test case: `find /i "9780061120084"`
+
+      Expected: The resource with ISBN "9780061120084" is displayed, which should be "To Kill a Mockingbird" by Harper Lee.
+
+3. **Find by ID**
+    - Test case: `find /id 2`
+
+      Expected: The resource with ID "2" is displayed, which should be "To Kill a Mockingbird" by Harper Lee.
+
+4. **Find by Title**
+    - Test case: `find /t "The Great Gatsby"`
+
+      Expected: The resource titled "The Great Gatsby" is displayed.
+
+5. **Combining Filters**
+    - Test case: `find /a "F. Scott Fitzgerald" /t "The Great Gatsby"`
+
+      Expected: Resources that match either the title "The Great Gatsby" or the author "F. Scott Fitzgerald" are displayed.
+
+
+### Add Resource Feature
+
+The `add` feature is responsible for processing user commands to add a new book to SysLib. It is facilitated by 
+the `AddCommand` component. It works with `Parser` and `Command` components to parse and validate the user input. 
+The new book is stored internally in `resourceList` as a `Book`. 
+
+`add` has seven options:
+- add /id [id] /t [title] /a [author] /tag [tag] /i [isbn]
+- add /id [id] /t [title] /a [author] /tag [tag] /i [isbn] _/g [genre] /s [status]_
+
+#### Implementation
+
+It implements the following operations:
+
+- `ADDCOMMAND#parseArgument(statement: String)` -- Parses the input command to extract relevant information.
+- `ADDCOMMAND#validate(statement: String, values: String[])` -- Validates the input statement to ensure that it is valid.
+- `ADDCOMMAND#createBook(values: String[])` -- Creates a new book based on the parsed and validated values.
+
+#### Example Usage Scenario
+
+Step 1. The user inputs the command: `add /id 0005 /t Frankenstein /a Mary Shelley /i FKS0001 /tag B /g Gothic, Fiction`
+
+Step 2. The `UI` component forwards the input to `SYSLIB`, which in turn passes it to the `PARSER`.
+
+Step 3. The `PARSER` processes the command and determines that it contains a valid key (`add`). It then calls 
+`ADDCOMMAND#execute(statement: String, this: Parser)` with the input command.
+
+Step 4. The `ADDCOMMMAND` component receives the command and performs the following operations:
+- Calls `ADDCOMMAND#parseArgument(statement: String)` to extract values for ID, title, author, ISBN, tag, and genres.
+- Calls `ADDCOMMAND#validate(statement: String, values: String[])` to ensure the validity of the input command.
+
+Step 5. The `COMMAND` component processes the input command to ensure that it meets the required format and constraints.
+It prepares the argument values for further processing.
+
+Step 6. Since the `tag` argument in the input command indicates that it is a book, the `ADDCOMMAND` determines that the
+key is equal to `b` (ignoring case). It then creates a new `Book` object using the parsed values (title, ISBN, author, 
+genres, ID).
+
+Step 7. The newly created book is forwarded to the `PARSER` to be added to the `resourceList`.
```

Comment:
A well-written, detailed explanation of the process. Great for a future developer!